core/modules/widgets/transclude.js

```diff
@@ -41,28 +41,45 @@ TranscludeWidget.prototype.execute = function() {
 	this.collectAttributes();
 	this.collectStringParameters();
 	this.collectSlotFillParameters();
-	// Get the target text and parse tree nodes that we are transcluding
-	var target = this.getTransclusionTarget(),
-		parseTreeNodes;
-	this.sourceText = target.text;
-	this.parserType = target.type;
-	this.parseAsInline = target.parseAsInline;
+	// Determine whether we're being used in inline or block mode
+	var parseAsInline = !this.parseTreeNode.isBlock;
+	if(this.transcludeMode === "inline") {
+		parseAsInline = true;
+	} else if(this.transcludeMode === "block") {
+		parseAsInline = false;
+	}
 	// Set 'thisTiddler'
 	this.setVariable("thisTiddler",this.transcludeTitle);
+	var parseTreeNodes, target;
 	// Process the transclusion according to the output type
 	switch(this.transcludeOutput || "text/html") {
 		case "text/html":
-			// Return the parse tree nodes
+			// Return the parse tree nodes of the target
+			target = this.parseTransclusionTarget(parseAsInline);
+			this.sourceText = target.text;
+			this.parserType = target.type;
+			this.parseAsInline = target.parseAsInline;
 			parseTreeNodes = target.parseTreeNodes;
 			break;
 		case "text/raw":
 			// Just return the raw text
+			target = this.getTransclusionTarget();
+			this.sourceText = target.text;
+			this.parserType = target.type;
 			parseTreeNodes = [{type: "text", text: this.sourceText}];
 			break;
 		default:
-			// text/plain
-			var plainText = this.wiki.renderText("text/plain",this.parserType,this.sourceText,{parentWidget: this});
-			parseTreeNodes = [{type: "text", text: plainText}];
+			// "text/plain" is the plain text result of wikifying the text
+			target = this.parseTransclusionTarget();
+			this.sourceText = target.text;
+			this.parserType = target.type;
```

Comment:
The 2 lines above seem to be the same code for all 3 states. They may be moved above the switch statement. Or is it intended to be that way?

Reply:
I think `this.parseAsInline` should also initialized and moved outside the cases.

Reply:
> The 2 lines above seem to be the same code for all 3 states. They may be moved above the switch statement. Or is it intended to be that way?

Yes that can be made a bit more efficient.

> I think `this.parseAsInline` should also initialized and moved outside the cases.

I don't think so. It is not required on all code paths